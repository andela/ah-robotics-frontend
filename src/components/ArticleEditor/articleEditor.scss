.right-column {
  text-align: right;
}

.cke_contents .cke_bottom {
  float: left;
  width: 100%;
}

.cke_reset {
  min-height: 400px;
  width: 100%;
}

.editor-input {
  width: 60%;
  margin-bottom: 1em;
  margin-top: 1em;
}

.editor-error {
  color: red;
}
.success-message{
color: #00d0a0;
}
.ui.green.button {
  margin: 1em;
  margin-right: -0.2em;
}