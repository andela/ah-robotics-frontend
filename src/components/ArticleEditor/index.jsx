import {
  Button, Container, Form, Grid,
} from 'semantic-ui-react';
import CKEditor from 'react-ckeditor-component';
import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import './articleEditor.scss';

const ArticleEditor = ({
 onInputChange, onEditorChange, handleArticlePost, createArticle, body,
}) => (
  <Container>
    <Fragment>
      <Grid>
        <Grid.Row>
          <Grid.Column width={12}>
            <p
              className="editor-error"
            >
              {(createArticle.errors) && createArticle.errors.title && 'Title is required'}
            </p>
            <Form.Input
              fluid
              name="title"
              placeholder="Title"
              transparent
              className="editor-input"
              onChange={onInputChange}
              maxLength={80}
            />
            <p
              className="editor-error"
            >
              {(createArticle.errors) && createArticle.errors.description && 'Description is required'}
            </p>
            <Form.Input
              fluid
              name="description"
              placeholder="Description"
              transparent
              className="editor-input"
              onChange={onInputChange}
              required
              maxLength={80}
            />
            <p className="success-message">
              {(createArticle.isPosted) && 'Article posted successfully'}
            </p>
          </Grid.Column>
          <Grid.Column width={4} className="right-column">
            <Button
              type="submit"
              basic
              color="green"
              content="Post"
              position="right"
              onClick={handleArticlePost}
              loading={createArticle.isPosting}
            />
          </Grid.Column>
        </Grid.Row>
      </Grid>
      <div>
        {' '}
        <p className="editor-error">{(createArticle.errors) && createArticle.errors.body && 'Body cannot be empty'}</p>
      </div>
      <CKEditor content={body} events={{ change: onEditorChange }} />
    </Fragment>
  </Container>
);
ArticleEditor.propTypes = {
  onInputChange: PropTypes.func.isRequired,
  onEditorChange: PropTypes.func.isRequired,
  handleArticlePost: PropTypes.func.isRequired,
  createArticle: PropTypes.shape({}).isRequired,
  body: PropTypes.string.isRequired,
};
export default ArticleEditor;
