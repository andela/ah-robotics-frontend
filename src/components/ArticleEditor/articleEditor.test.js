import React from 'react';
import ArticleEditor from './index';

const handleArticlePost = jest.fn();
const onEditorChange = jest.fn();
const onInputChange = jest.fn();
const wrapper = shallow(<ArticleEditor
  createArticle={{}}
  handleArticlePost={handleArticlePost}
  onEditorChange={onEditorChange}
  body=""
  onInputChange={onInputChange}
/>);

describe('test article editor component', () => {
  it('it render article editor component', () => {
    expect(wrapper.length).toBe(1);
  });
  it('it should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
