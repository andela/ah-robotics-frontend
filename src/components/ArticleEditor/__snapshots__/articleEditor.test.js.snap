// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test article editor component it should match snapshot 1`] = `
<Container>
  <Grid>
    <GridRow>
      <GridColumn
        width={12}
      >
        <p
          className="editor-error"
        />
        <FormInput
          as={[Function]}
          className="editor-input"
          control={[Function]}
          fluid={true}
          maxLength={80}
          name="title"
          onChange={[MockFunction]}
          placeholder="Title"
          transparent={true}
        />
        <p
          className="editor-error"
        />
        <FormInput
          as={[Function]}
          className="editor-input"
          control={[Function]}
          fluid={true}
          maxLength={80}
          name="description"
          onChange={[MockFunction]}
          placeholder="Description"
          required={true}
          transparent={true}
        />
        <p
          className="success-message"
        />
      </GridColumn>
      <GridColumn
        className="right-column"
        width={4}
      >
        <Button
          as="button"
          basic={true}
          color="green"
          content="Post"
          onClick={[MockFunction]}
          position="right"
          type="submit"
        />
      </GridColumn>
    </GridRow>
  </Grid>
  <div>
     
    <p
      className="editor-error"
    />
  </div>
  <CKEditor
    activeClass=""
    config={Object {}}
    content=""
    events={
      Object {
        "change": [MockFunction],
      }
    }
    isScriptLoaded={false}
    scriptUrl="https://cdn.ckeditor.com/4.6.2/standard/ckeditor.js"
  />
</Container>
`;
